### PR TITLE
Added information about JAVA

### PR DESCRIPTION
**Pull request by abisz**
I thought it would be interesting knowing a few more things about Java's platform independence.
I also added the use case of using JAVA in backend development.

### CODE REVIEW

**Comment by abhisekp** [comment]
# :+1: LGTM

@alayek what are your thoughts on the additions?

**Comment by alayek** [comment]
@abisz thanks for this PR! Thing is, that article was meant to be only a placeholder for all things Java on our Wiki. Like our [Python](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Python) home page; which has link to other Python pages.

We didn't initially include those description, because our target audience for this particular article was someone trying to learn Java. We thought it would overwhelm them to have so much info on what bytecode is, what machine code is etc.

File: Java.md [hunk 1]
```diff
@@ -9,8 +9,8 @@
 
 Java was conceptualized back in [1991](http://web.archive.org/web/20050420081440/http://java.sun.com/features/1998/05/birthday.html), but it was too advanced for its original purpose of helping broadcast cable news better. Finally, in 1995, first public release of Java 1.0 was made by Sun Microsystems.
 
-This language has been around a long time, and is known for its battle-hardened reliability. The [JVM](https://en.wikipedia.org/wiki/Java_virtual_machine) lets you write platform independent Java code.
+This language has been around a long time, and is known for its battle-hardened reliability. The [JVM](https://en.wikipedia.org/wiki/Java_virtual_machine) lets you write platform independent Java code. This is achieved by first compiling the source code to bytecode, which is later translated to corresponding machine code. The big advantage of this procedure is that bytecode runs inside the JVM, which is responsible for compiling it into matching machine code during runtime. So every time a new operating system comes out, Oracle has to write a JVM and every JAVA application can theoretically run on it.
```
Comment (alayek) [inline Java.md]: Like to what bytecode is - maybe some Wikipedia article or some article that explains well.

Comment (alayek) [inline Java.md]: Link to what machine code is.

Comment (alayek) [inline Java.md]: > So every time a new operating system comes out, Oracle has to write a JVM and every JAVA application can theoretically run on it.

That's a funny way to put it! But no, new OS, at least at a kernel level, don't come out that often.

**Comment by abisz** [comment]
@alayek I'm not offended if you think this isn't the appropriate place (or time) to go into details. I didn't put a lot of time into these changes and have no problem with closing this PR, just say the word :)

**Comment by alayek** [comment]
No, it's okay! Just link those two properly, I would merge your changes.

**Comment by abisz** [comment]
@alayek Ok I made the changes, thanks for the suggestions. Next time i'll write an issue before starting to work on something, promise!

**Comment by Rafase282** [comment]
# :+1: LGTM

**Comment by abhisekp** [comment]
@abisz You may Chat with us in <kbd><a href="https://gitter.im/FreeCodeCamp/Wiki"> <b>FreeCodeCamp/Wiki</b></a></kbd>
Thanks for your contribution. :tada: :smiley: 

Let it keep coming. Love your work :heartpulse:

**Comment by alayek** [comment]
# LGTM :+1: 

:heart_decoration: your work!

**Pull request merged**